### DISCUSSION TITLE
recording server setup

### DISCUSSION THREAD

**Discussion by sfrankel** [jamulussoftware/jamulus#3414] [General]
Hello, I'd like to use Jamulus to record but it's not working for some reason.

I created a fresh new AWS Lightsail Ubuntu server and followed all current setup instructions for a private Jamulus server, including configuring systemd files for the headless server and sending SIGUSR signals. I created a directory for recording under /srv to avoid homedir issues. I can connect to the server, systemctl status output shows that it's receiving the signals, but the client Jamulus app does not show a red banner and no new recording files are written to my /srv/recordings directory. 

Here's a snippet of my systemd file:
...
ExecStart=/usr/bin/jamulus-headless -s -n \
    --log /var/log/jamulus/jamulus.log \
    --welcomemessage "Welcome here!" \
    --numchannels 16 \
    --recording /srv/recordings/ \
    --norecord
...
    
Here's systemctl output showing that signals are sent and received:
systemd[1]: jamulus-headless.service: Sent signal SIGUSR2 to main process 1532 (jamulus-headles) on client request.
jamulus[1532]: OnHandledSignal: 12
systemd[1]: jamulus-headless.service: Sent signal SIGUSR1 to main process 1532 (jamulus-headles) on client request.
jamulus[1532]: OnHandledSignal: 10

Suggestions? TIA!

**Comment by dcorson-ticino-com** [comment]
Were users actually connected to the server when you tested the recording?
I don't know all the details of the recording on/off automation, but normally Jamulus only records when users are actually connected to the server.

**Reply by sfrankel** [reply]
One user (me) was connected. The client app registered audio in its "LED" display. No files were written on the server that I'm aware of, certainly not in the /srv/recordings/ dir.

I want to stress that this is a clean install on a newly created AWS:Lightsail instance, following all installation and systemd instructions, with default naming conventions.

Surely there's something obvious I'm overlooking...

**Reply by dcorson-ticino-com** [reply]
The other obvious thing: are you sure that the rights for your recordings directory allow Jamulus to write there ?

**Reply by rdica** [reply]
@sfrankel Could you paste the complete systemd unit file, and output of `sudo ls -ld {/srv,/srv/recordings}`

**Reply by sfrankel** [reply]
> The other obvious thing: are you sure that the rights for your recordings directory allow Jamulus to write there ?

Yep, that was the obvious thing I asked for. Issue resolved with a simple `chmod 777`

Thanks!

**Reply by ann0see** [reply]
Better chown the directory to the correct user (jamulus). Usually you don't want to have a directory world writable.

**Reply by sfrankel** [reply]
> Better chown the directory to the correct user (jamulus). Usually you don't want to have a directory world writable.

Yes, excellent suggestion. Thanks!
